src/msg.rs

```diff
@@ -354,40 +354,33 @@ pub enum QueryMsg {
 }
 
 impl QueryMsg {
-    pub fn get_validation_params(&self) -> (Vec<&HumanAddr>, ViewingKey) {
+    pub fn get_validation_params(&self) -> (Vec<&String>, ViewingKeyObj) {
```

Comment:
The validations are being done in each query function

Reply:
Exactly, I think you can reduce code reuse this way

Reply:
I think it will beat the purpose because permit_queries() also uses query_balance(), query_transfers(),... 
And it doesn't need to perform a viewing key validation.
So if we'll validate the addresses in get_validation_params(), we still won't be able to remove the address validation inside query_balance(), query_transfers(),...
Unless we add another different validation function inside permit_queries(),
but what will we save then?

Reply:
I see your point.
For the record, I think the current implementation in the toolkit needs to be improved (e.g. permit::validate() should return `StdResult<Addr>` and not `StdResult<String>`, but for now I guess we can keep it this way.

Reply:
For permits it's better to use String as it can give us compatibility with non-Secret addresses in the future. Addr validates the HRP AFAIK

Reply:
@Cashmaney String is used for all inputs (transactions & queries).
But I'll remove the address validation - on the strings received in permit queries.
That's what you meant right?

Reply:
@Cashmaney the validation is just 
```rust
return humanize(canonicalize(human_addr)) == human_addr
```